packages/backend/src/payment-method/local/service.ts

```diff
@@ -0,0 +1,289 @@
+          retryable: false
+        }
+      )
+    }
+    debitAmountValue = converted
+  } else {
+    throw new PaymentMethodHandlerError('Received error during local quoting', {
+      description: 'No value provided to get quote from',
+      retryable: false
+    })
+  }
+
+  if (debitAmountValue <= BigInt(0)) {
+    throw new PaymentMethodHandlerError('Received error during local quoting', {
+      description: 'debit amount of local quote is non-positive',
+      retryable: false
+    })
+  }
+
+  if (receiveAmountValue <= BigInt(0)) {
+    throw new PaymentMethodHandlerError('Received error during local quoting', {
+      description: 'receive amount of local quote is non-positive',
+      retryable: false,
+      code: PaymentMethodHandlerErrorCode.QuoteNonPositiveReceiveAmount
+    })
+  }
+
+  return {
+    receiver: options.receiver,
+    walletAddress: options.walletAddress,
+    estimatedExchangeRate:
+      Number(receiveAmountValue) / Number(debitAmountValue),
+    debitAmount: {
+      value: debitAmountValue,
+      assetCode: options.walletAddress.asset.code,
+      assetScale: options.walletAddress.asset.scale
+    },
+    receiveAmount: {
+      value: receiveAmountValue,
+      assetCode: options.receiver.assetCode,
+      assetScale: options.receiver.assetScale
+    },
+    additionalFields: {}
+  }
+}
+
+async function pay(
+  deps: ServiceDependencies,
+  options: PayOptions
+): Promise<void> {
+  const { outgoingPayment, receiver, finalReceiveAmount, finalDebitAmount } =
+    options
+
+  const incomingPaymentId = receiver.incomingPayment.id.split('/').pop()
+  if (!incomingPaymentId) {
+    throw new PaymentMethodHandlerError('Received error during local payment', {
+      description: 'Failed to parse incoming payment on receiver',
+      retryable: false
+    })
+  }
+  const incomingPayment = await deps.incomingPaymentService.get({
+    id: incomingPaymentId
+  })
+  if (!incomingPayment) {
+    throw new PaymentMethodHandlerError('Received error during local payment', {
+      description: 'Incoming payment not found from receiver',
+      retryable: false
+    })
+  }
+
+  // TODO: remove incoming state check? perhaps only applies ilp account middleware where its checking many different things
+  if (incomingPayment.state === IncomingPaymentState.Pending) {
+    try {
+      await deps.accountingService.createLiquidityAccount(
+        incomingPayment,
+        LiquidityAccountType.INCOMING
+      )
+    } catch (err) {
+      if (!(err instanceof AccountAlreadyExistsError)) {
+        deps.logger.error(
+          { incomingPayment, err },
+          'Failed to create liquidity account for local incoming payment'
+        )
+        throw new PaymentMethodHandlerError(
+          'Received error during local payment',
+          {
+            description:
+              'Unknown error while trying to create liquidity account',
+            retryable: false
+          }
+        )
+      }
+    }
+  }
```

Comment:
was not sure if we needed this `incomingPayment.state === IncomingPaymentState.Pending` check here (pulled from connector) - it should probably always be pending right? maybe we check and error if not ... just wasnt quite sure why it made sense in the connector